File: client/src/app/model/ispit.ts
import {IspitniRok} from "./ispitniRok";
import {Predmet} from "./predmet";
import {StudijskiProgram} from "./studijskiProgram";

export interface Ispit {
  id: number,
  pocetakIspita: Date,
  krajIspita: Date,
  ispitniRok: IspitniRok,
  predmet: Predmet
  studijskiProgram: StudijskiProgram
}

File: server/src/main/java/rs/ac/singidunum/novisad/server/model/RealizacijaPredmeta/Ispit.java
package rs.ac.singidunum.novisad.server.model.RealizacijaPredmeta;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.cglib.core.Local;
import rs.ac.singidunum.novisad.server.model.fakultet.StudijskiProgram;
import rs.ac.singidunum.novisad.server.model.predmet.Predmet;

import java.sql.Time;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.time.OffsetDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Ispit {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private LocalDateTime pocetakIspita;
    private LocalDateTime krajIspita;
    @ManyToOne
    private IspitniRok ispitniRok;
    @ManyToOne
    private Predmet predmet;
    @ManyToOne
    private StudijskiProgram studijskiProgram;
}
